tools/clang/test/HLSLFileCheck/validation/verify_output_complete_dce.ll

```diff
@@ -0,0 +1,114 @@
+target triple = "dxil-ms-dx"
+
+%dx.types.NodeHandle = type { i8* }
+%dx.types.NodeInfo = type { i32, i32 }
+%dx.types.NodeRecordHandle = type { i8* }
+%dx.types.NodeRecordInfo = type { i32, i32 }
+%struct.loadStressRecord.0 = type { [3 x i32], [3 x i32] }
+
+@"\01?loadStressTemp@@3PAIA" = external addrspace(3) global [128 x i32], align 4
+
+define void @loadStress_16() {
+  %1 = call i32 @dx.op.flattenedThreadIdInGroup.i32(i32 96)  ; FlattenedThreadIdInGroup()
+  %2 = call %dx.types.NodeHandle @dx.op.createNodeOutputHandle(i32 247, i32 0)  ; CreateNodeOutputHandle(MetadataIdx)
+  %3 = call %dx.types.NodeHandle @dx.op.annotateNodeHandle(i32 249, %dx.types.NodeHandle %2, %dx.types.NodeInfo { i32 6, i32 24 })  ; AnnotateNodeHandle(node,props)
+  %4 = call %dx.types.NodeRecordHandle @dx.op.allocateNodeOutputRecords(i32 238, %dx.types.NodeHandle %3, i32 1, i1 true)  ; AllocateNodeOutputRecords(output,numRecords,perThread)
+  %5 = call %dx.types.NodeRecordHandle @dx.op.annotateNodeRecordHandle(i32 251, %dx.types.NodeRecordHandle %4, %dx.types.NodeRecordInfo { i32 38, i32 24 })  ; AnnotateNodeRecordHandle(noderecord,props)
+  %6 = urem i32 %1, 3
+  %7 = add nuw nsw i32 %6, 1
+  %8 = call %struct.loadStressRecord.0 addrspace(6)* @dx.op.getNodeRecordPtr.struct.loadStressRecord.0(i32 239, %dx.types.NodeRecordHandle %5, i32 0)  ; GetNodeRecordPtr(recordhandle,arrayIndex)
+  %9 = getelementptr inbounds %struct.loadStressRecord.0, %struct.loadStressRecord.0 addrspace(6)* %8, i32 0, i32 0, i32 0
+  store i32 %7, i32 addrspace(6)* %9, align 4
+  %10 = getelementptr inbounds %struct.loadStressRecord.0, %struct.loadStressRecord.0 addrspace(6)* %8, i32 0, i32 0, i32 1
+  store i32 1, i32 addrspace(6)* %10, align 4
+  %11 = getelementptr inbounds %struct.loadStressRecord.0, %struct.loadStressRecord.0 addrspace(6)* %8, i32 0, i32 0, i32 2
+  store i32 1, i32 addrspace(6)* %11, align 4
+  %12 = load i32, i32 addrspace(3)* getelementptr inbounds ([128 x i32], [128 x i32] addrspace(3)* @"\01?loadStressTemp@@3PAIA", i32 0, i32 0), align 4, !tbaa !22
+  %13 = getelementptr inbounds %struct.loadStressRecord.0, %struct.loadStressRecord.0 addrspace(6)* %8, i32 0, i32 1, i32 0
+  store i32 %12, i32 addrspace(6)* %13, align 4
+  %14 = load i32, i32 addrspace(3)* getelementptr inbounds ([128 x i32], [128 x i32] addrspace(3)* @"\01?loadStressTemp@@3PAIA", i32 0, i32 1), align 4, !tbaa !22
+  %15 = getelementptr inbounds %struct.loadStressRecord.0, %struct.loadStressRecord.0 addrspace(6)* %8, i32 0, i32 1, i32 1
+  store i32 %14, i32 addrspace(6)* %15, align 4
+  %16 = load i32, i32 addrspace(3)* getelementptr inbounds ([128 x i32], [128 x i32] addrspace(3)* @"\01?loadStressTemp@@3PAIA", i32 0, i32 2), align 4, !tbaa !22
+  %17 = getelementptr inbounds %struct.loadStressRecord.0, %struct.loadStressRecord.0 addrspace(6)* %8, i32 0, i32 1, i32 2
+  store i32 %16, i32 addrspace(6)* %17, align 4
```

Comment:
For the purpose of checking for the elimination of these through DCE, I don't think most of the code here is useful.  While it was generated from some HLSL, a lot of code could probably be deleted without making it invalid.  You can probably delete all instructions not involved in getting the valid handle value for the one outputComplete call you don't want deleted (`%6` through the last `store`).  Then you can delete unused function declarations too.